### DISCUSSION TITLE
MapBox Styling: Is icon-image & fill-pattern possible?

### DISCUSSION THREAD

**Discussion by Sverre-Iversen** [ldproxy/ldproxy#726] [Configuration] [answered]
Hi, I need some examples for the use of icon-image and fill-pattern for styling of features with customized (local) images. Is this a limitation in ldproxy or is this possible?

**Comment by cportele** [comment, marked as answer]
You need to use and reference a sprite for images in Mapbox/Maplibre Styles. See, for example, https://maplibre.org/maplibre-gl-js-docs/style-spec/sprite/.

**Reply by Sverre-Iversen** [reply]
I have tried the sprite at the demo.ldproxy.net-path:
https://demo.ldproxy.net/daraa/resources/sprites is working both for symbol and pattern 👍 

I copied the json- and the png-file to a local server from https://demo.ldproxy.net/daraa/resources/sprites to a local server https://geo.ngu.no/kart/images/sprites/daraa/sprites
Its neither working for symbol or pattern **- any clues?**

(I still have a problem with my own sprites, so the demo.ldproxy.net-sprite should be a good starting point I thought)

**Reply by cportele** [reply]
It looks as if geo.ngu.no does not support CORS, so the browser will not fetch the files. You should be able to see the blocked requests in the developer view in the browser.

**Reply by Sverre-Iversen** [reply]
Thanks, so it is a CORS-problem. I think we can set this right.

**Reply by mspyhm2001** [reply]
Hi everyone. I am using ldproxy version 3.6.
I want to display the desired icon in the MapBox format in the map style.
I am an amateur in this field. Can you please guide me more than the content here?

**Reply by cportele** [reply]
@mspyhm2001 - You need to create and use a sprite in the style. For more information on sprites and how to create them see https://maplibre.org/maplibre-style-spec/sprite/.

**Reply by mspyhm2001** [reply]
@cportele Thank you Clemens,
Yes, I had seen this link.
Is there an example or done project that I can use?

**Reply by cportele** [reply]
A [sample sprite](https://github.com/ldproxy/demo/tree/main/resources/api-resources/daraa) is available in the ["daraa" API configuration](https://github.com/ldproxy/demo/blob/main/entities/instances/features/withTiles/services/daraa.yml). The styles are [here](https://github.com/ldproxy/demo/tree/main/values/maplibre-styles/daraa).

**Reply by mspyhm2001** [reply]
thank you. I have my own icons in svg format.
How can I convert them to sprite so that I have a png-sheet file and an associated json file?

**Reply by cportele** [reply]
I do not have experience with this, but using Google I see that there are several tools for this.

**Reply by mspyhm2001** [reply]
I found these 2 tools. Looks good

https://www.facetstudios.com/sprite-generator

https://www.finalparsec.com/tools/sprite_sheet_maker

**Comment by Sverre-Iversen** [comment]
Thanks 🙂

________________________________
